stale

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+set -Eeuo pipefail
+
+unknown_command() {
+	echo "Unknown command: $*"
+	exit 1
+}
+
+print_help() {
+	echo 'https://github.com/shiftstack/shiftstack-ci'
+	echo
+	echo 'Prints the IDs of the stale resources.'
+	echo
+	echo 'Usage:'
+	echo "$0 <resource_type>"
+	echo
+	echo '"resource_type" can be any openstack resource type.'
+	echo 'Resources are identified as stale if they were last updated more than 5 hours ago.'
+	echo
+	echo 'Examples:'
+	echo "$0 port"
```

Comment:
I like this script but... isn't it a bit dangerous? Someone might have the bright idea to pass the output to `xargs openstack <resource> delete` without measuring the consequences. But some stale resources shouldn't be deleted. See for example the `project-identifier` security group.

Reply:
Can you think of other user-deletable resources that we don’t want to delete? If no, I’d be tempted to just add the project identifiers to a blocklist

Reply:
To be completely honest with you, ‘xargs os delete’ was exactly what I had in mind 😇

Reply:
There's a lot more than just the project identifier, for example:
- all the shared images (and especially not rhcos one)
- the default security group and associated rules
- the external network and maybe also the private one (not sure if it's used) and associated ports
- probably many more

Assuming that the installer tags all resources it creates, it would be safer to filter out the resources that don't have a tag in the expected format.

Reply:
OK I understand that this needs attention. Let’s see how the tag discussions go with Product. In the meantime, better not pipe to xargs indeed

Reply:
I have added a `-u` tag for printing the date the resource was last updated. With this, I was able to prune only resources created in the last couple weeks, without touching resources "there to stay". The usage of this tool, while it is requiring the due attention, has recently proven very handy for pruning volumes.